mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -457,30 +457,81 @@ def _get_deleted_runs(self):
             return [run_id[0] for run_id in run_ids]
 
     def log_metric(self, run_id, metric):
-        _validate_metric(metric.key, metric.value, metric.timestamp, metric.step)
-        is_nan = math.isnan(metric.value)
-        if is_nan:
-            value = 0
-        elif math.isinf(metric.value):
-            #  NB: Sql can not represent Infs = > We replace +/- Inf with max/min 64b float value
-            value = 1.7976931348623157e308 if metric.value > 0 else -1.7976931348623157e308
-        else:
-            value = metric.value
+        self.log_metrics(run_id, [metric], auto_new=False)
+
+    def log_metrics(self, run_id, metrics, auto_new=True):
+        if metrics is None:
+            return
+
         with self.ManagedSessionMaker() as session:
-            run = self._get_run(run_uuid=run_id, session=session)
-            self._check_run_is_active(run)
-            # ToDo: Consider prior checks for null, type, metric name validations, ... etc.
-            logged_metric, just_created = self._get_or_create(
-                model=SqlMetric, run_uuid=run_id, key=metric.key, value=value,
-                timestamp=metric.timestamp, step=metric.step, session=session, is_nan=is_nan)
-            # Conditionally update the ``latest_metrics`` table if the logged metric  was not
-            # already present in the ``metrics`` table. If the logged metric was already present,
-            # we assume that the ``latest_metrics`` table already accounts for its presence
-            if just_created:
-                self._update_latest_metric_if_necessary(logged_metric, session)
+            try:
+                run = self._get_run(run_uuid=run_id, session=session)
+                self._check_run_is_active(run)
+
+                metrics_per_key = {}
+
+                for metric in metrics:
+
+                    _validate_metric(metric.key, metric.value, metric.timestamp, metric.step)
+                    is_nan = math.isnan(metric.value)
+                    if is_nan:
+                        value = 0
+                    elif math.isinf(metric.value):
+                        #  NB: Sql can not represent Infs = > We replace +/- Inf
+                        #  with max/min 64b float value
+                        value = 1.7976931348623157e308 if metric.value > 0 \
+                            else -1.7976931348623157e308
+                    else:
+                        value = metric.value
+
+                    # ToDo: Consider prior checks for null, type, metric name validations, ... etc.
+
+                    # Entries will be added to session. If conflicts are detection during saving,
+                    # metrics are inserted one after another using _get_or_create
```

Comment:
I am not sure I understand this comment. Also unclear why `auto_new` is required

Reply:
I improved the comments and added some details.